### PR TITLE
schema cache: fix wrong timestamp in schema cache

### LINKED ISSUE #46325: schema cache should try to correct previous wrong timestamp
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

when get schema encounters transient error when gets its timestamp, the timestamp of the schema is zero
https://github.com/pingcap/tidb/blob/v7.3.0/domain/domain.go#L223
it makes the cache useless because the timestamp is zero and it won't recover because the next success reload won't correct the timestamp until another ddl happens, the reload will exit at without correct the timestamp
https://github.com/pingcap/tidb/blob/v7.3.0/domain/domain.go#L227

### 2. What did you expect to see? (Required)
transient error should not prevent schema cache being useful forever

### 3. What did you see instead (Required)
schema cache is not useable

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by hihihuhu**
### What problem does this PR solve?
correct the wrong timestamp in schema cache to make it useful

Issue Number: close #46325 

Problem Summary: 

### What is changed and how it works?
try to update the timestamp of the existing schema cache when the timestamp is fetched

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -230,6 +230,10 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	}
 
 	if is := do.infoCache.GetByVersion(neededSchemaVersion); is != nil {
+		if schemaTs > 0 {
+			// try to insert here as well to correct the schemaTs if previous is wrong
+			do.infoCache.Insert(is, uint64(schemaTs))
```
Comment (xhebox) [inline domain/domain.go:235]: I think we should internalize `schemaTs > 0` into `infocache.Insert`. Rest LGTM

Comment (hihihuhu) [reply domain/domain.go]: removed, the insert already overwrite the timestamp only if it is not zero

**Review by xhebox — APPROVED** [review]

**Pull request merged**